src/main/java/org/wso2/carbon/hash/provider/pbkdf2/PBKDF2HashProvider.java

```diff
@@ -0,0 +1,166 @@
+
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import java.util.Map;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+/**
+ * This class contains the implementation for the PBKDF2 hashing algorithm.
+ */
+public class PBKDF2HashProvider implements HashProvider {
+
+    private static final Log log = LogFactory.getLog(PBKDF2HashProvider.class);
+
+    @Override
+    public byte[] getHash(String value, String salt, Map<String, Object> metaProperties)
+            throws HashProviderException {
+
+        validateEmptyValue(value);
+        int iterationCount = getCount(metaProperties);
+        int dkLen = getDerivedKeyLength(metaProperties);
+        String pseudoRandomFunction = getPseudoRandomFunction(metaProperties);
+        return pbkdf2HashCalculation(value, salt, iterationCount, dkLen, pseudoRandomFunction);
+    }
+
+    @Override
+    public String getAlgorithm() {
+
+        return Constants.DEFAULT_HASHING_ALGORITHM;
+    }
+
+    /**
+     * This method is responsible for validating the value.
+     *
+     * @param value The value which needs to be hashed.
+     * @throws HashProviderException The custom exception which is thrown at validating the value.
+     */
+    private void validateEmptyValue(String value) throws HashProviderException {
+
+        if (StringUtils.isBlank(value)) {
+            throw new HashProviderException(
+                    ErrorMessage.ERROR_CODE_EMPTY_VALUE.getDescription(),
+                    Constants.IDENTITY_HASH_PROVIDER_PBKDF2_ERROR_PREFIX +
+                            ErrorMessage.ERROR_CODE_EMPTY_VALUE.getCode());
+        }
+    }
+
+    /**
+     * This method is responsible for validating and getting iteration count from the metaProperties.
+     *
+     * @param metaProperties The Map which has the properties which needs for PBKDF2 hashing algorithm.
+     * @return the iteration count which needs to be get.
+     */
+    private int getCount(Map<String, Object> metaProperties) {
+
+        if (metaProperties.get(Constants.ITERATION_COUNT_KEY) == null) {
+            return Constants.DEFAULT_ITERATION_COUNT;
+        }
+        return (int) metaProperties.get(Constants.ITERATION_COUNT_KEY);
+    }
+
+    /**
+     * This method is responsible for validating and getting derived key length from the metaProperties.
+     *
+     * @param metaProperties The Map which has the properties which needs for PBKDF2 hashing algorithm.
+     * @return the derived key length which needs to be get.
+     */
+    private int getDerivedKeyLength(Map<String, Object> metaProperties) {
+
+        if (metaProperties.get(Constants.DERIVED_KEY_LENGTH_KEY) == null) {
+            return Constants.DEFAULT_DERIVED_KEY_LENGTH;
+        }
+        return (int) metaProperties.get(Constants.DERIVED_KEY_LENGTH_KEY);
+    }
```

Comment:
fix the method name, comments and formatting according to the above method review comments.